File: core/src/main/java/uk/lobsterdoodle/namepicker/Log.java
package uk.lobsterdoodle.namepicker;

public class Log {
    static private Logger logger = intoTheVoid();

    public static void install(Logger logger) {
        Log.logger = logger;
    }

    public static Logger as() {
        return logger;
    }

    private static Logger intoTheVoid() {
        return new Logger() {
            @Override
            public void v(String tag, String msg) { }

            @Override
            public void d(String tag, String msg) { }

            @Override
            public void i(String tag, String msg) { }

            @Override
            public void w(String tag, String msg) { }

            @Override
            public void e(String tag, String msg, Throwable ex) { }

            @Override
            public void wtf(String tag, String msg, Throwable ex) { }
        };
    }
}

File: core/src/main/kotlin/uk/lobsterdoodle/namepicker/Log.kt
package uk.lobsterdoodle.namepicker

object Log {
    private var logger = noOp()

    fun install(logger: Logger) {
        Log.logger = logger
    }

    private fun noOp(): Logger {
        return object : Logger {
            override fun v(tag: String, msg: String) {}
            override fun d(tag: String, msg: String) {}
            override fun i(tag: String, msg: String) {}
            override fun w(tag: String, msg: String) {}
            override fun e(tag: String, msg: String, ex: Throwable) {}
            override fun wtf(tag: String, msg: String, ex: Throwable) {}
        }
    }
}
